subprojects/testing-jvm/src/main/java/org/gradle/api/internal/tasks/testing/junit/JUnitSpec.java

```diff
@@ -22,11 +22,13 @@
     private final Set<String> includeCategories;
     private final Set<String> excludeCategories;
     private final Set<String> includedTests;
+    private final String requestClass;
 
-    public JUnitSpec(Set<String> includeCategories, Set<String> excludeCategories, Set<String> includedTests) {
+    public JUnitSpec(Set<String> includeCategories, Set<String> excludeCategories, Set<String> includedTests, String requestClass) {
```

Comment:
It would be better to avoid breaking the signature of this constructor as some people may be using the internal API. Adding an overload constructor would be the way to go. I also remove all modification done to `JUnitTestClassProcessorTest.java`.